src/classes/UTIL_DMLService_TEST.cls

```diff
@@ -0,0 +1,697 @@
+
+        UTIL_CustomSettingsFacade.getErrorSettingsForTests(new Error_Settings__c(Respect_Duplicate_Rule_Settings__c = false));
+        Error_Settings__c errorSettings = UTIL_CustomSettingsFacade.getErrorSettings();
+        UTIL_DMLService.insertRecord(errorSettings);
+
+        Account acct = new Account(Name='Blue Cloud', BillingCity='San Francisco');
+        UTIL_DMLService.insertRecord(acct);
+        List<Account> retrievedAccts = [SELECT Id, Name FROM Account];
+        System.assertEquals('Blue Cloud', retrievedAccts[0].Name);
+
+        Account dupeAcct = new Account(Name='Blue Cloud', BillingCity='San Francisco');
+        Database.SaveResult sr = UTIL_DMLService.insertRecord(dupeAcct, false); // AllOrNone flag set to false to check if the Duplicate Rule's Action is set to Block. 
+        if (!sr.isSuccess()) {
+            for (Database.Error duplicateError : sr.getErrors()) {
+                System.assertEquals(ERROR_STATUS_CODE_DUPLICATES_DETECTED, String.valueOf(duplicateError.getStatusCode()));
+                Datacloud.DuplicateResult duplicateResult = ((Database.DuplicateError)duplicateError).getDuplicateResult();
+                // If isAllowSave is false, the running Org has the Duplicate Rule's Action set to Block so end test.
+                if (!duplicateResult.isAllowSave()) return;
+            }
+        }
+
+        retrievedAccts = [SELECT Id, Name FROM Account];
+        System.assertEquals(NUMBER_OF_RECORDS_ONE, retrievedAccts.size());
+
+        errorSettings.Respect_Duplicate_Rule_Settings__c = true;
+        UTIL_DMLService.updateRecord(errorSettings);
+
+        UTIL_DMLService.insertRecord(dupeAcct);
+
+        retrievedAccts = [SELECT Id, Name FROM Account];
+        System.assertEquals(NUMBER_OF_RECORDS_TWO, retrievedAccts.size());
+
+        Account dupeAcctUpdate = new Account(Name='Blue Cloud Update', BillingCity='San Francisco');
+        UTIL_DMLService.insertRecord(dupeAcctUpdate);
+        retrievedAccts = [SELECT Id, Name FROM Account];
+        System.assertEquals(NUMBER_OF_RECORDS_THREE, retrievedAccts.size());
+
+        dupeAcctUpdate.Name = 'Blue Cloud';
+        UTIL_DMLService.updateRecord(dupeAcctUpdate);
+
+        retrievedAccts = [SELECT Id, Name FROM Account];
+        for(Account eachAcct : retrievedAccts) {
+            System.assertEquals('Blue Cloud', eachAcct.Name);
+        }
+    }
+
+
+    /*********************************************************************************************************
+    * @description Tests the NPSP Setting Respect Duplicate Rule Settings for duplicate Contacts
+    *   Records To Be Saved set to true 
+    * verify:
+    *    Duplicate Contacts detected and allowed when the Respect Duplicate Rule Setting set to true
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRespectDuplicateRuleSettingsForContacts() {
```

Comment:
consider doing the same with this routine, so we can test both updates and upserts.